src/OpenTelemetry.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -767,7 +771,7 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(FunctionID function
 {
     auto _ = trace::Stats::Instance()->JITCompilationStartedMeasure();
 
-    if (!is_attached_ || !is_safe_to_block)
+    if (!is_attached_ || !is_safe_to_block || use_dotnet_startuphook_bootstrapper)
```

Comment:
Is the StartupHook the first method called for compilation?

Reply:
In the mongodb sample app, the startup hook is not the first method called for compilation but it is one of the first. The first method was the static .ctor for the program entry point. Here's a snippet from the beginning of the profiler logs to show the sequence.

```
11/16/21 08:02:26.278 PM [17912|14132] [info] JIT Inlining is enabled.
11/16/21 08:02:26.278 PM [17912|14132] [info] NGEN is disabled.
11/16/21 08:02:26.278 PM [17912|14132] [info] Profiler attached.
11/16/21 08:02:26.288 PM [17912|14132] [debug] AssemblyLoadStarted: 7766520
11/16/21 08:02:26.288 PM [17912|14132] [debug] ModuleLoadStarted: 11157520
11/16/21 08:02:26.295 PM [17912|14132] [debug] ModuleLoadFinished: 11157520 mscorlib AppDomain 1963837352 EE Shared Assembly Repository | IsNGEN = false | IsDynamic = false | IsResource = false
11/16/21 08:02:26.304 PM [17912|14132] [info] COR library: mscorlib 4.0.0
11/16/21 08:02:26.304 PM [17912|14132] [debug] AssemblyLoadFinished: 7766520 0
11/16/21 08:02:26.304 PM [17912|14132] [debug] AssemblyLoadFinished: AssemblyName=mscorlib AssemblyVersion=4.0.0.0
11/16/21 08:02:26.321 PM [17912|14132] [debug] AppDomainCreationStarted: 1963838200
11/16/21 08:02:26.321 PM [17912|14132] [debug] AppDomainCreationFinished: 1963838200 hrStatus=0
11/16/21 08:02:26.321 PM [17912|14132] [debug] AppDomainCreationStarted: 7547136
11/16/21 08:02:26.321 PM [17912|14132] [debug] AppDomainCreationFinished: 7547136 hrStatus=0
11/16/21 08:02:26.321 PM [17912|14132] [debug] AppDomainCreationStarted: 1963837352
11/16/21 08:02:26.321 PM [17912|14132] [debug] AppDomainCreationFinished: 1963837352 hrStatus=0
11/16/21 08:02:26.351 PM [17912|14132] [debug] AssemblyLoadStarted: 7766704
11/16/21 08:02:26.351 PM [17912|14132] [debug] ModuleLoadStarted: 11223108
11/16/21 08:02:26.351 PM [17912|14132] [debug] ModuleLoadFinished: 11223108 Samples.MongoDB AppDomain 7547136 DefaultDomain | IsNGEN = false | IsDynamic = false | IsResource = false
11/16/21 08:02:26.351 PM [17912|14132] [debug] ModuleLoadFinished stored metadata for 11223108 Samples.MongoDB AppDomain 7547136 DefaultDomain
11/16/21 08:02:26.351 PM [17912|14132] [debug] AssemblyLoadFinished: 7766704 0
11/16/21 08:02:26.351 PM [17912|14132] [debug] AssemblyLoadFinished: AssemblyName=Samples.MongoDB AssemblyVersion=0.0.0.0
11/16/21 08:02:26.377 PM [17912|14132] [debug] JITCompilationStarted: function_id=11224684 token=100663304 name=Samples.MongoDB.Program..cctor()
11/16/21 08:02:26.377 PM [17912|14132] [info] JITCompilationStarted: Startup hook registered in function_id=11224684 token=100663304 name=Samples.MongoDB.Program..cctor(), assembly_name=Samples.MongoDB app_domain_id=7547136 domain_neutral=0
11/16/21 08:02:26.377 PM [17912|14132] [debug] Using existing corlib reference: mscorlib
11/16/21 08:02:26.387 PM [17912|14132] [debug] AssemblyLoadStarted: 84910464
11/16/21 08:02:26.387 PM [17912|14132] [debug] ModuleLoadStarted: 11225340
11/16/21 08:02:26.396 PM [17912|14132] [debug] ModuleLoadFinished: 11225340 System.Diagnostics.DiagnosticSource AppDomain 7547136 Samples.MongoDB.exe | IsNGEN = false | IsDynamic = false | IsResource = false
11/16/21 08:02:26.396 PM [17912|14132] [debug] ModuleLoadFinished skipping module by pattern: 11225340 System.Diagnostics.DiagnosticSource
11/16/21 08:02:26.396 PM [17912|14132] [debug] AssemblyLoadFinished: 84910464 0
11/16/21 08:02:26.396 PM [17912|14132] [debug] AssemblyLoadFinished: AssemblyName=System.Diagnostics.DiagnosticSource AssemblyVersion=6.0.0.0
11/16/21 08:02:26.406 PM [17912|14132] [debug] AssemblyLoadStarted: 84901624
11/16/21 08:02:26.406 PM [17912|14132] [debug] ModuleLoadStarted: 11230060
11/16/21 08:02:26.406 PM [17912|14132] [debug] ModuleLoadFinished: 11230060 OpenTelemetry.ClrProfiler.Managed.Loader AppDomain 7547136 Samples.MongoDB.exe | IsNGEN = false | IsDynamic = false | IsResource = false
11/16/21 08:02:26.406 PM [17912|14132] [info] ModuleLoadFinished: OpenTelemetry.ClrProfiler.Managed.Loader loaded into AppDomain 7547136 Samples.MongoDB.exe
11/16/21 08:02:26.406 PM [17912|14132] [debug] AssemblyLoadFinished: 84901624 0
11/16/21 08:02:26.406 PM [17912|14132] [debug] AssemblyLoadFinished: AssemblyName=OpenTelemetry.ClrProfiler.Managed.Loader AssemblyVersion=0.0.1.0
```

Reply:
I see, it makes sense.